src/test/java/org/opensearch/ad/ml/EntityColdStarterTests.java

```diff
@@ -458,4 +509,258 @@ public void testEmptyDataRange() throws InterruptedException {
+                AnomalyDetectorPlugin.AD_THREAD_POOL_NAME
+            );
+
+            entityColdStarter = new EntityColdStarter(
+                clock,
+                threadPool,
+                stateManager,
+                AnomalyDetectorSettings.NUM_SAMPLES_PER_TREE,
+                AnomalyDetectorSettings.MULTI_ENTITY_NUM_TREES,
+                AnomalyDetectorSettings.TIME_DECAY,
+                numMinSamples,
+                AnomalyDetectorSettings.MAX_SAMPLE_STRIDE,
+                AnomalyDetectorSettings.MAX_TRAIN_SAMPLE,
+                interpolator,
+                searchFeatureDao,
+                AnomalyDetectorSettings.DEFAULT_MULTI_ENTITY_SHINGLE,
+                AnomalyDetectorSettings.THRESHOLD_MIN_PVALUE,
+                AnomalyDetectorSettings.THRESHOLD_MAX_RANK_ERROR,
+                AnomalyDetectorSettings.THRESHOLD_MAX_SCORE,
+                AnomalyDetectorSettings.THRESHOLD_NUM_LOGNORMAL_QUANTILES,
+                AnomalyDetectorSettings.THRESHOLD_DOWNSAMPLES,
+                AnomalyDetectorSettings.THRESHOLD_MAX_SAMPLES,
+                featureManager,
+                AnomalyDetectorSettings.HOURLY_MAINTENANCE,
+                AnomalyDetectorSettings.MAX_SMALL_STATES,
+                checkpoint,
+                settings,
+                seed
+            );
+
+            // create labelled data
+            MultiDimDataWithTime dataWithKeys = LabelledAnomalyGenerator
+                .getMultiDimData(
+                    dataSize + detector.getShingleSize() - 1,
+                    50,
+                    100,
+                    5,
+                    seed,
+                    baseDimension,
+                    false,
+                    trainTestSplit,
+                    delta,
+                    anomalyIndependent
+                );
+
+            long[] timestamps = dataWithKeys.timestampsMs;
+            double[][] data = dataWithKeys.data;
+            when(clock.millis()).thenReturn(timestamps[trainTestSplit - 1]);
+
+            doAnswer(invocation -> {
+                ActionListener<Entry<Optional<Long>, Optional<Long>>> listener = invocation.getArgument(2);
+                listener.onResponse(new SimpleImmutableEntry<>(Optional.of(timestamps[0]), Optional.of(timestamps[trainTestSplit - 1])));
+                return null;
+            }).when(searchFeatureDao).getEntityMinMaxDataTime(any(), any(), any());
+
+            doAnswer(invocation -> {
+                List<Entry<Long, Long>> ranges = invocation.getArgument(1);
+                List<Optional<double[]>> coldStartSamples = new ArrayList<>();
+
+                Collections.sort(ranges, new Comparator<Entry<Long, Long>>() {
+                    @Override
+                    public int compare(Entry<Long, Long> p1, Entry<Long, Long> p2) {
+                        return Long.compare(p1.getKey(), p2.getKey());
+                    }
+                });
+                for (int j = 0; j < ranges.size(); j++) {
```

Comment:
Do the ranges we get here have actual values to allow us to mock the actual expected results by mapping them to the timestamps?

Reply:
yes